ccm_web/client/src/pages/BulkSectionCreate.tsx

```diff
@@ -0,0 +1,534 @@
+  }
+
+  useEffect(() => {
+    parseUpload(file)
+  }, [file])
+
+  const resetPageState = (): void => {
+    setPageState({ state: BulkSectionCreatePageState.Upload, schemaInvalidation: [], rowInvalidations: [] })
+  }
+
+  const handleSchemaError = (schemaInvalidations: SectionsSchemaInvalidation[]): void => {
+    setPageState({ state: BulkSectionCreatePageState.InvalidUpload, schemaInvalidation: schemaInvalidations, rowInvalidations: [] })
+  }
+
+  const handleRowLevelInvalidationError = (invalidations: SectionsRowInvalidation[]): void => {
+    setPageState({ state: BulkSectionCreatePageState.InvalidUpload, schemaInvalidation: [], rowInvalidations: invalidations })
+  }
+
+  const handleParseSuccess = (sectionNames: string[]): void => {
+    setSectionNames(sectionNames)
+    setPageState({ state: BulkSectionCreatePageState.Confirm, schemaInvalidation: [], rowInvalidations: [] })
+  }
+
+  const parseUpload = (file: File|undefined): void => {
+    if (file === undefined) {
+      resetPageState()
+      return
+    }
+    file.text().then(t => {
+      let lines = t.split(/[\r\n]+/).map(line => { return line.trim() })
+      // An empty file will resultin 1 line
+      if (lines.length > 0 && lines[0].length === 0) {
+        lines = lines.slice(1)
+      }
+
+      const schemaInvalidations: SectionsSchemaInvalidation[] = []
+
+      const sectionNameHeaderValidator: SectionsSchemaValidator = new SectionNameHeaderValidator()
+      schemaInvalidations.push(...sectionNameHeaderValidator.validate(lines))
+
+      if (schemaInvalidations.length !== 0) {
+        handleSchemaError([{ error: 'Error processing file', type: InvalidationType.Error }])
+        return
+      }
+
+      if (hasHeader(lines)) {
+        lines = lines.slice(1)
+      }
+
+      const rowInvalidations: SectionsRowInvalidation[] = []
+
+      const duplicateNamesInFileValidator: SectionRowsValidator = new DuplicateSectionInFileSectionRowsValidator()
+      rowInvalidations.push(...duplicateNamesInFileValidator.validate(lines))
+
+      const duplicatesNamesInCanvasValidator: DuplicateExistingSectionRowsValidator = new DuplicateExistingSectionRowsValidator()
+      rowInvalidations.push(...duplicatesNamesInCanvasValidator.validate(lines))
+
+      if (rowInvalidations.length !== 0) {
+        handleRowLevelInvalidationError(rowInvalidations)
+        return
+      }
+
+      handleParseSuccess(lines)
+    }).catch(e => {
+      // TODO Not sure how to produce this error in real life
+      handleSchemaError([{ error: 'Error processing file', type: InvalidationType.Error }])
+    })
+  }
+
+  const renderUploadHeader = (): JSX.Element => {
+    return <div className={classes.uploadHeader}>
+      <Typography variant='h6'>Upload your CSV File</Typography>
+      <br/>
+      <Typography><strong>Requirement:</strong> Your file should include one section name per line</Typography>
+      <Typography>View a <Link href='#'>CSV File Example</Link></Typography>
```

Comment:
I added a popover for this.  It's simpler than the example here b/c the table seemed unnecessary since it's just 1 "column" but I'm happy to make it look fancier somehow

Reply:
@chrisrrowland can you share a screenshot for it? Let Janel/Melinda answer that question,

Reply:
![image](https://user-images.githubusercontent.com/17617998/119857081-4507f300-bee1-11eb-911c-7e623cb6274b.png)

Reply:
@melindakraft @janelilr , is that sample file for bulk creation look ok?

Reply:
I have a follow up question first. When someone creates a section name, does it need to be a specific naming convention? For example: Course name term with section number (ALA 225 Summer Section 001).

Reply:
I think you can name them whatever you want? @pushyamig 
![image](https://user-images.githubusercontent.com/17617998/119870183-1133ca00-beef-11eb-8ae4-45f1c0b45bec.png)

Reply:
Generally they do follow a pattern for convenience I guess as below. But they can chose any name they want. Usually the sample file is for the format of the CSV file and we don't dictate the pattern of section name. The subaccount admins has been doing this for years, so they will be pretty clear what they want to name their sections.
![Screen Shot 2021-05-27 at 1 30 16 PM](https://user-images.githubusercontent.com/8579775/119871046-104f6800-bef0-11eb-9a1d-88d11bc36053.png)

Reply:
I can make it automatically generate example section names based on the course name if we want (as long as that info is made available to the UI)

Reply:
@chrisrrowland regarding your comment automatically generating section name based on course name and I don't think if we want to add more logic to that Unless we are making the pop-up for showing sample example file as downloadable file.

Reply:
The course name has to come from the back end, I think.  The UX logic is pretty trivial.
I passed arguments "Test" and 3 to the component.

![image](https://user-images.githubusercontent.com/17617998/119876868-58718900-bef6-11eb-81a3-e531384624a4.png)

Reply:
Another possible way to display the example would be:
<img width="159" alt="Screen Shot 2021-05-27 at 2 12 48 PM" src="https://user-images.githubusercontent.com/46663379/119877095-a090ab80-bef6-11eb-8def-3060bb013d28.png">

This visual gives a little more context on what the csv file submitting would like.

Reply:
Is this the new requirement?  Is this supposed to be a UI  component that looks like excel?  A screenshot?  Are we going to accompany the example with instructions to export to csv from excel? Many questions!

Reply:
The Readme will have instruction how to make a CSV file. But People who are using this feature know that very well. No need elaborate description.

I don't think there is requirement to make look like an excel file or screenshot, I think what Janel was proposing is having some sort of vertical and horizontal line indication. Do What is simplest the goal is to tell then that it needs a header and section name that's all. Nothing fancy. We will add things in Readme for a fuller picture

Reply:
If someone could please tell me exactly what the requirement is that would be great.  What is simplest is of course for me to leave it as is :)

Reply:
OK, we will stick with what we have. I don't feel we need to go more deeper conversation than this. I will make this as resolved.

Reply:
> I think the simplest would be add row number and horizontal line to give a feel of file to convey information one item i.e section name per row in the file. If that needs complicated logic then I will stick with what you have for now. It's conveys the message we want to give!

There is no complicated logic unless we want to construct a table component.  If it's a screenshot, it's just static content and we can no longer have dynamic section names (unless we want to get really complicated)

This is getting weird to me b/c this is supposed to be an example of the file, which is most certainly not a spreadsheet and doesn't contain row numbers.  I get if we're trying to convey that it's "CSV"  (which it's not) like they are used to, but at the same time having an "example file' that isn't really an example of what the file contains is goofy!